packages/server/src/sync/api.ts

```diff
@@ -1,69 +1,38 @@
-import { SyncRequest } from '@botpress/messaging-base'
-import { Request, Router, Response } from 'express'
+import { Response } from 'express'
 import Joi from 'joi'
-import _ from 'lodash'
-import yn from 'yn'
-import { methodNotAllowed } from '../base/api/utils'
-import { Auth } from '../base/auth/auth'
+import { ApiManager } from '../base/api-manager'
+import { ClientApiRequest } from '../base/auth/client'
 import { ChannelService } from '../channels/service'
-import { ClientTokenService } from '../client-tokens/service'
-import { ClientService } from '../clients/service'
 import { makeSyncRequestSchema } from './schema'
 import { SyncService } from './service'
 
 export class SyncApi {
-  private force!: boolean
   private schema!: Joi.ObjectSchema
 
-  constructor(
-    private syncs: SyncService,
-    private clients: ClientService,
-    private clientTokens: ClientTokenService,
-    private channels: ChannelService
-  ) {}
+  constructor(private syncs: SyncService, private channels: ChannelService) {}
 
-  setup(router: Router, auth: Auth) {
-    this.force = !!yn(process.env.SPINNED)
+  setup(router: ApiManager) {
     this.schema = makeSyncRequestSchema(this.channels.list())
-
-    router
-      .route('/sync')
-      .post(auth.public.auth(this.sync.bind(this)))
-      .all(methodNotAllowed('POST'))
+    router.post('/sync', this.schema, this.sync.bind(this))
   }
 
-  async sync(req: Request, res: Response) {
-    const channelsWithoutEnabled: { [channelName: string]: Object | undefined } = {}
-    for (const [channelName, channelConfig] of Object.entries<typeof channelsWithoutEnabled>(
-      req.body?.channels || {}
-    )) {
-      if (channelConfig?.enabled === false) {
-        continue
-      }
+  async sync(req: ClientApiRequest, res: Response) {
+    const body = { ...req.body, channels: this.removeDisabledChannels(req.body.channels || {}) }
+    const { value } = this.schema.validate(body)
 
-      channelsWithoutEnabled[channelName] = _.omit(channelConfig, ['enabled'])
-    }
-    const bodyWithoutEnabled = { ...(req.body || {}), channels: channelsWithoutEnabled }
-
-    const { error, value } = this.schema.validate(bodyWithoutEnabled)
-    if (error) {
-      return res.status(400).send(error.message)
-    }
+    const result = await this.syncs.sync(req.clientId, value)
+    res.send(result)
+  }
 
-    const sync = value as SyncRequest
+  private removeDisabledChannels(body: any) {
```

Comment:
I would be easy here to use some kind of interface instead of any since all channels share the enable property!

Reply:
I use any here since there can techically be anything in there, since we use the stripUnknown feature.